src/wp-includes/media.php

```diff
@@ -2097,10 +2097,10 @@ function _wp_post_thumbnail_class_filter_remove( $attr ) {
  *     @type string $caption    The caption text.
  *     @type string $class      Additional class name(s) added to the caption container.
  * }
- * @param string $content Shortcode content.
+ * @param string $content Shortcode content. Defaults to an empty string.
  * @return string HTML content to display the caption.
  */
-function img_caption_shortcode( $attr, $content = null ) {
+function img_caption_shortcode( $attr, $content = '' ) {
```

Comment:
Note for review: investigate how shortcode is handled when there's no content provided. Is it `null` or an empty string?